src/script-runner.js

```diff
@@ -16,24 +17,72 @@ window.starknet = starknet
 window.chai = chai
 window.ethers = ethers
 
+const scriptReturns = {} // keep track of modules exported values
 window.require = (module) => {
-  if(window[module]) return window[module]
+  if (window[module]) return window[module] // library
+  if (window.__execPath__ && scriptReturns[window.__execPath__]) return scriptReturns[window.__execPath__][module] // module exported values
   else throw new Error(`${module} module require is not supported by Remix IDE`)
 }
 
 class CodeExecutor extends PluginClient {
-  execute (script) {
+  async execute (script, filePath, fromPath) {
+    filePath = filePath || 'scripts/script.ts'
+    const paths = filePath.split('/')
+    paths.pop()
+    fromPath = fromPath || paths.join('/') // get current execcution context path
     if (script) {
       try {
-        (new Function(script))()
+        script = ts.transpileModule(script, { moduleName: filePath, filePath,
+        compilerOptions: {
+         target: ts.ScriptTarget.ES2015,
+         module: ts.ModuleKind.CommonJS 
+        }});
+        script = script.outputText;
+
+        // extract all the "require", execute them and store the returned values.
+        const regexp = /require\("(.*?)"\)/g
+        const array = [...script.matchAll(regexp)];
+
+        for (const regex of array) {
+          let file = regex[1]
+          let absolutePath = file
+          if (file.startsWith('./')) {
+            absolutePath = paths.join('/') + file.substring(1)
+          }
+          if (!scriptReturns[fromPath]) scriptReturns[fromPath] = {}
+          scriptReturns[fromPath][file] = await this.executeFile(absolutePath, fromPath)
+        }
+
+        // execute the script
+        script = `const exports = {};
+                  window.__execPath__ = "${fromPath}"
+                  ${script}; 
+                  return exports`
+        const returns = (new Function(script))()
         if (mocha.suite && ((mocha.suite.suites && mocha.suite.suites.length) || (mocha.suite.tests && mocha.suite.tests.length))) mocha.run()
+        return returns
       } catch (e) {
         this.emit('error', {
           data: [e.message]
         })
       }
     }
   }
+
+  async _resolveFile (fileName) {
```

Comment:
With this implementation, i will not be able to have two files of the same name with different extensions in the same directory. E.g `script.js` and `script.ts`. Is this the intended implementation?

Reply:
I think this is ok if we set the `ts` file to be run instead of `js`